Remove Tensor.h, TensorMethods.h from src/core.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25031 Remove Tensor.h, TensorMethods.h from src/core.**

This is a major source of merge conflicts, and AFAICT isn't necessary anymore (it may have been necessary for some mobile build stuff in the past).

Differential Revision: [D16963597](https://our.internmc.facebook.com/intern/diff/D16963597)

gchanan committed:
Remove Tensor.h, TensorMethods.h from src/core.

This is a major source of merge conflicts, and AFAICT isn't necessary anymore (it may have been necessary for some mobile build stuff in the past).

zou3519 reviewed (approved):
lgtm. As we discussed offline, it might be better to leave `core/{Tensor,TensorMethods}.h` checked in (but ignored) for now to avoid merge conflicts and finally delete them at a later date.

gchanan committed:
Update on "Remove Tensor.h, TensorMethods.h from src/core."


This is a major source of merge conflicts, and AFAICT isn't necessary anymore (it may have been necessary for some mobile build stuff in the past).

Differential Revision: [D16963597](https://our.internmc.facebook.com/intern/diff/D16963597)

gchanan committed:
Update on "Remove Tensor.h, TensorMethods.h from src/core."


This is a major source of merge conflicts, and AFAICT isn't necessary anymore (it may have been necessary for some mobile build stuff in the past).

Differential Revision: [D16963597](https://our.internmc.facebook.com/intern/diff/D16963597)

gchanan commented:
It looks like there are two issues here:
1) windows builds aren't getting the correct directory separator
2) dependencies still don't seem to be tracked correctly, perhaps because it's necessary that we do codegen before we generate core now.

gchanan committed:
Update on "Remove Tensor.h, TensorMethods.h from src/core."


This is a major source of merge conflicts, and AFAICT isn't necessary anymore (it may have been necessary for some mobile build stuff in the past).

Differential Revision: [D16963597](https://our.internmc.facebook.com/intern/diff/D16963597)

gchanan committed:
Update on "Remove Tensor.h, TensorMethods.h from src/core."


This is a major source of merge conflicts, and AFAICT isn't necessary anymore (it may have been necessary for some mobile build stuff in the past).

Differential Revision: [D16963597](https://our.internmc.facebook.com/intern/diff/D16963597)

gchanan committed:
Update on "Remove Tensor.h, TensorMethods.h from src/core."


This is a major source of merge conflicts, and AFAICT isn't necessary anymore (it may have been necessary for some mobile build stuff in the past).

Differential Revision: [D16963597](https://our.internmc.facebook.com/intern/diff/D16963597)

gchanan committed:
Update on "Remove Tensor.h, TensorMethods.h from src/core."


This is a major source of merge conflicts, and AFAICT isn't necessary anymore (it may have been necessary for some mobile build stuff in the past).

Differential Revision: [D16963597](https://our.internmc.facebook.com/intern/diff/D16963597)

gchanan commented:
This has been completed by @ezyang.
